bitcoin/src/bip32.rs

```diff
@@ -256,108 +368,272 @@ impl serde::Serialize for ChildNumber {
         u32::from(*self).serialize(serializer)
     }
 }
+/// `normal_derivation_path` module with inner methods.
+pub mod normal_derivation_path {
+    use super::*;
 
-/// Trait that allows possibly failable conversion from a type into a
-/// derivation path
-pub trait IntoDerivationPath {
-    /// Converts a given type into a [`DerivationPath`] with possible error
-    fn into_derivation_path(self) -> Result<DerivationPath, Error>;
-}
+    /// A [`DerivationPath`] that can only contain [`NormalChildNumber`] (i.e. no hardened indexes).
+    #[derive(Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+    pub struct NormalDerivationPath(Vec<NormalChildNumber>);
 
-/// A BIP-32 derivation path.
-#[derive(Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
-pub struct DerivationPath(Vec<ChildNumber>);
+    #[cfg(feature = "serde")]
+    internals::serde_string_impl!(
+        NormalDerivationPath,
+        "a BIP-32 derivation path containing normal numbers only"
+    );
+
+    impl NormalDerivationPath {
+        /// The Master (empty) path (i.e. the master key).
+        #[allow(dead_code)]
+        const MASTER: Self = NormalDerivationPath(Vec::new());
```

Comment:
This should be `pub`, the `allow` is wrong.

Reply:
"resolved"